ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTest.java

```diff
@@ -0,0 +1,531 @@
+          r -> testDriver.pipeInput(
+              new ConsumerRecordFactory<>(
+                  Serdes.String().serializer(),
+                  r.topic.getSerdeSupplier().getSerializer(schemaRegistryClient)
+              ).create(r.topic.name, r.key, r.value, r.timestamp)
+          )
+      );
+    }
+
+    @SuppressWarnings("unchecked")
+    void verifyOutput(final TopologyTestDriver testDriver,
+                      final SchemaRegistryClient schemaRegistryClient) {
+      try {
+        outputRecords.forEach(
+            r -> OutputVerifier.compareKeyValueTimestamp(
+                testDriver.readOutput(
+                    r.topic.name, r.keyDeserializer(),
+                    r.topic.getSerdeSupplier().getDeserializer(schemaRegistryClient)
+                ),
+                r.key(),
+                r.value,
+                r.timestamp)
+        );
+      } catch (AssertionError assertionError) {
+        throw new AssertionError("Query name: "
+            + name
+            + " in file: " + testPath
+            + " failed due to: "
+            + assertionError.getMessage());
+      }
+    }
+
+    void initializeTopics(final KsqlEngine ksqlEngine) {
+      for (Topic topic : topics) {
+        ksqlEngine.getTopicClient().createTopic(topic.getName(), 1, (short) 1);
+        if (topic.getSchema() != null) {
+          try {
+            ksqlEngine.getSchemaRegistryClient().register(
+                topic.getName() + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX, topic.getSchema());
+          } catch (Exception e) {
+            throw new RuntimeException(e);
+          }
+        }
+      }
+    }
+  }
+
+  private static TopologyTestDriver buildStreamsTopology(final Query query, final KsqlEngine ksqlEngine,
+                                                         final Properties streamsProperties) {
+    final List<QueryMetadata> queries = new ArrayList<>();
+    query.statements().forEach(
+        q -> queries.addAll(ksqlEngine.buildMultipleQueries(q, Collections.emptyMap()))
+    );
+    return new TopologyTestDriver(queries.get(queries.size() - 1).getTopology(),
+        streamsProperties,
+        0);
+  }
+
+  static List<String> findTests(final String dir) throws IOException {
+    final List<String> tests = new ArrayList<>();
+    try (final BufferedReader reader =
+             new BufferedReader(
+                 new InputStreamReader(EndToEndEngineTest.class.getClassLoader().
+                     getResourceAsStream(dir)))) {
+
+      String test;
+      while ((test = reader.readLine()) != null) {
+        if (test.endsWith(".json")) {
+          tests.add(test);
+        }
+      }
+    }
+    return tests;
+  }
+
+  static void shouldBuildAndExecuteQuery(Query query) {
+    final MetaStore metaStore = new MetaStoreImpl(new InternalFunctionRegistry());
+    final Map<String, Object> config = new HashMap<String, Object>() {{
+      put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
```

Comment:
I think this should be the address of the embedded kafka?

Reply:
The test doesn't actually run any broker. The config is just there to placate streams and admin client.